swatch-picker/open-swatch-picker.js

```diff
@@ -0,0 +1,2204 @@
+                </div>
+                
+                <div swatch="">
+                <button aria-label="Copy Rose 5" style="background:var(--rose-5)">
+                    &nbsp;
+                </button>
+                <span>Rose 5</span>
+                </div>
+                
+                <div swatch="">
+                <button aria-label="Copy Rose 6" style="background:var(--rose-6)">
+                    &nbsp;
+                </button>
+                <span>Rose 6</span>
+                </div>
+                
+                <div swatch="">
+                <button aria-label="Copy Rose 7" style="background:var(--rose-7)">
+                    &nbsp;
+                </button>
+                <span>Rose 7</span>
+                </div>
+                
+                <div swatch="">
+                <button aria-label="Copy Rose 8" style="background:var(--rose-8)">
+                    &nbsp;
+                </button>
+                <span>Rose 8</span>
+                </div>
+                
+                <div swatch="">
+                <button aria-label="Copy Rose 9" style="background:var(--rose-9)">
+                    &nbsp;
+                </button>
+                <span>Rose 9</span>
+                </div>
+                
+                <div swatch="">
+                <button aria-label="Copy Rose 10" style="background:var(--rose-10)">
+                    &nbsp;
+                </button>
+                <span>Rose 10</span>
+                </div>
+                
+                <div swatch="">
+                <button aria-label="Copy Rose 11" style="background:var(--rose-11)">
+                    &nbsp;
+                </button>
+                <span>Rose 11</span>
+                </div>
+                
+                <div swatch="">
+                <button aria-label="Copy Rose 12" style="background:var(--rose-12)">
+                    &nbsp;
+                </button>
+                <span>Rose 12</span>
+                </div>
+                
+            </div>
+
+        </main>
+    </dialog>
+`
+export class OpenSwatchPicker extends HTMLElement {
+    static define(tagName = "open-swatch-picker") {
+        customElements.define(tagName, this)
+    }
+    shadowRoot = this.attachShadow({ mode: "open" });
+
+    constructor(){
+        super();
+        this._internals = this.attachInternals();
+    }
+
+    _value = '';
+    dialog = null;
+    hexValue = '';
+    hexValue = '';
+    rgbValue = '';
+
+    get value() {
+        return this._value;
+    }
+
+    set value(newValue) {
+        if (this._value !== newValue) {
+            this._value = newValue;
+            this.onValueChange(newValue); // Call the watcher function
+        }
+    }
+
+    get outputType() {
+        return this.getAttribute('output-type');
+    }
+
+    get showOutputLabel() {
+        return this.getAttribute('show-output-label');
+    }
+
+    oklchToOklab(oklch) {
```

Comment:
These could be private:


```suggestion
    #oklchToOklab(oklch) {
```

Alternatively hoist them out of the class, or even better in a separate file. One day CSSOM should be fully featured enough that the browser will be able to do this stuff for us.

(Same for all other colour conversion methods).